DICOM/src/Microsoft.Health.Anonymizer.Common/RedactFunction.cs

```diff
@@ -0,0 +1,155 @@
+                    try
+                    {
+                        return RedactAge(decimal.Parse(inputString)).ToString();
+                    }
+                    catch
+                    {
+                        throw new AnonymizerException(AnonymizerErrorCode.RedactFailed, "The input value is not a numeric age value.");
+                    }
+
+                case AnonymizerValueTypes.PostalCode:
+                    return RedactPostalCode(inputString);
+                default:
+                    return null;
+            }
+        }
+
+        public uint? Redact(uint value, AnonymizerValueTypes valueType)
+        {
+            return valueType switch
+            {
+                AnonymizerValueTypes.Age => RedactAge(value),
+                _ => null,
+            };
+        }
+
+        public decimal? Redact(decimal value, AnonymizerValueTypes valueType)
+        {
+            return valueType switch
+            {
+                AnonymizerValueTypes.Age => RedactAge(value),
+                _ => null,
+            };
+        }
+
+        public DateTimeOffset? Radact(DateTimeOffset dateTime)
+        {
+            EnsureArg.IsNotNull<DateTimeOffset>(dateTime, nameof(dateTime));
+
+            if (_redactSetting.EnablePartialDatesForRedact && !DateTimeUtility.IndicateAgeOverThreshold(dateTime))
+            {
+                return new DateTimeOffset(dateTime.Year, 1, 1, 0, 0, 0, dateTime.Offset);
+            }
+
+            return null;
+        }
+
+        public DateTimeObject Redact(DateTimeObject dateObject)
+        {
+            EnsureArg.IsNotNull(dateObject, nameof(dateObject));
+
+            if (_redactSetting.EnablePartialDatesForRedact && !DateTimeUtility.IndicateAgeOverThreshold(dateObject.DateValue))
+            {
+                dateObject.DateValue = new DateTimeOffset(dateObject.DateValue.Year, 1, 1, 0, 0, 0, (bool)dateObject.HasTimeZone ? dateObject.DateValue.Offset : default);
+                return dateObject;
+            }
+
+            return null;
+        }
+
+        public AgeObject Redact(AgeObject age)
+        {
+            EnsureArg.IsNotNull(age, nameof(age));
+
+            return RedactAge(age.AgeInYears()) == null ? null : age;
+        }
+
+        private string RedactDateTime(string dateTimeString, string dateTimeFormat)
+        {
+            DateTimeOffset date = DateTimeUtility.ParseDateTimeString(dateTimeString, dateTimeFormat);
+            return Radact(date)?.ToString(dateTimeFormat);
+        }
+
+        private uint? RedactAge(uint age)
+        {
+            return (uint?)RedactAge((decimal)age);
+        }
+
+        private decimal? RedactAge(decimal age)
+        {
+            if (_redactSetting.EnablePartialAgesForRedact)
+            {
+                if (age > Constants.AgeThreshold)
+                {
+                    return null;
+                }
+
+                return age;
+            }
+
+            return null;
```

Comment:
Looks this can be: `return _redactSetting.EnablePartialAgesForRedact && age <= Constants.AgeThreshold ? age : null;`

Reply:
Very concise👍 I posted a similar suggestion before. But original format has its own value. It's easier to maintain if there might be complex logic.

Reply:
Seems this expression not available in C#8.0